File: day1/day1_1.py
def day1():
    total = 0
    lines = []
    with open("./day1_1_input.txt") as f:
        for line in f:
            total += find_value(line)
    print(total)


def find_value(line):
    first, last = "", ""
    for i in range(len(line)):
        if line[i].isdigit():
            first = line[i]
            break

    for i in range(len(line) - 1, -1, -1):
        if line[i].isdigit():
            last = line[i]
            break

    return int(first + last)


day1()

File: day1/go/day1_1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"unicode"
)

func main() {
	file, err := os.Open("../day1_1_input.txt")
	if err != nil {
		log.Fatal("unable to open input file", err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)

	total := 0
	for scanner.Scan() {
		line := scanner.Text()
		lineRunes := []rune(line)

		numRunes := make([]rune, 0)
		for _, c := range lineRunes {
			if unicode.IsNumber(c) {
				numRunes = append(numRunes, c)
				break
			}
		}

		for i := len(lineRunes) - 1; i >= 0; i-- {
			if unicode.IsDigit(lineRunes[i]) {
				numRunes = append(numRunes, lineRunes[i])
				break
			}
		}

		numStr := string(numRunes)
		num, err := strconv.Atoi(numStr)
		if err != nil {
			log.Fatal("unable to perform number conversion", err)
		}

		total += int(num)
	}

	fmt.Println(total)
}
